### ISSUE TITLE
 The main thread stuck when initializing 2.8.1 onesignal

### ISSUE THREAD

**Issue by a455455b** [OneSignal/OneSignal-iOS-SDK#354]
Our app got stuck when it started . 
We paused and checked the threads
![_dc5208c3-3871-4e79-a130-565a8e06d852](https://user-images.githubusercontent.com/15962971/37821231-0d2acc48-2ebe-11e8-86af-f9bd6ba27d02.png)
We think it is a problem of semaphores

**Comment by Nightsd01** [comment]
Hi @a455455b 

We've seen this bug reported before but we've never been able to reproduce it.

Is there some reliable set of steps you can recommend to reproduce this issue, or does it only happen randomly for you?

**Comment by a455455b** [comment]
I can't reproduce this every time , About twice a week.
We also cannot know what your SDK is doing with semaphores.
But we don't think it's a good idea to wait on the main thread with a semaphore.
You should think about spin locks.

**Comment by Nightsd01** [comment]
@a455455b I am not sure what you mean? Our SDK is open source, you can view the code on this repo. 

Also, the code in question (`canTrack()`) isn’t using semaphores and isn’t blocking?

**Comment by a455455b** [comment]
emmmmmm @Nightsd01  Yeah ,Like you said ,I did initialize with such simple steps.

**Comment by Nightsd01** [comment]
Thank you for reporting this @a455455b , we have been investigating this for a while without much luck, but we will keep looking into it. 

Closing this issue because it is a duplicate of #248
